plugins/catalog-backend/src/modules/core/BuiltinKindsEntityProcessor.ts

```diff
@@ -298,6 +298,12 @@ export class BuiltinKindsEntityProcessor implements CatalogProcessor {
         RELATION_OWNED_BY,
         RELATION_OWNER_OF,
       );
+      doEmit(
+        domain.spec.subdomainOf,
+        { defaultKind: 'Domain', defaultNamespace: selfRef.namespace },
```

Comment:
it strikes me now - just thinking out loud, no action needed - that maybe we should have a `requireKind` in addition to (or instead of) `defaultKind` in the future here; no reason to let people write `subdomainOf: user:freben` :)

Reply:
Yes, that sounds sensible. @janogonzalez ?

Reply:
If we can have that in the future it would be great. Also a future validation at the schema level may come in handy?

Reply:
Hm not sure that can be done actually. The `spec` fields are shorthands and it'd be a bit awkward to properly express that the various possibilities of how to enter it are being used correctly. I think it's probably fine to leave it as just a string, and then let validation happen at runtime as part of the processing/validation steps.

As for the output data (in the `relations` block), those are generated output-only fields and don't need validation

Reply:
OK, we'll leave as-is for now then.